### PR TITLE
Feat (Rln relay): enabling signed tx

### PR DESCRIPTION
**Pull request by staheri14**
The rln-relay nodes require to sign their registration transactions to authorize the fund deduction from their ethereum accounts. This PR integrates the signing feature alongside with a CLI option to pass the Ethereum private key. 
Note that passing the Ethereum account private key is an MVP solution just for testing purposes and in no way is secure, however, it is tolerable for now as all the interactions take place on the Ethereum testnet Goerli with no actual fund involved.

Note that as part of this PR, the on-chain rln-relay tests needed to be updated to account for the Eth private key requirement. 

Next Step: https://github.com/status-im/nwaku/issues/819

### CODE REVIEW

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 1]
```diff
@@ -607,22 +611,27 @@ proc addAll*(rlnInstance: RLN[Bn256], list: seq[IDCommitment]): bool =
...
+  web3.defaultAccount = ethAccountAddress 
+  #  set the gas price twice the suggested price in order for the fast mining
+  # let gasPrice = int(await web3.provider.eth_gasPrice()) * 2
 
   # subscribe to the MemberRegistered events
   # TODO can do similarly for deletion events, though it is not yet supported
   discard await contractObj.subscribe(MemberRegistered, %*{"fromBlock": blockNumber, "address": contractAddress}) do(pubkey: Uint256, index: Uint256){.raises: [Defect], gcsafe.}:
     try:
       debug "onRegister", pubkey = pubkey, index = index
+      echo "onRegister",  pubkey,  index
```
Comment (jm-clius) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:633]: Lingering `echo`? :)

Comment (staheri14) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim]: Not any more :) https://github.com/status-im/nwaku/pull/1023/commits/cbff025143fc1417c620a5a6374c61575f290630

**Review by LNSD — APPROVED** [review]
LGTM ✅

**Review by kgrgpg — APPROVED** [review]
Looks good.

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim [hunk 2]
```diff
@@ -90,7 +90,7 @@ type MessageValidationResult* {.pure.} = enum
 # inputs of the membership contract constructor
 # TODO may be able to make these constants private and put them inside the waku_rln_relay_utils
 const
-  MEMBERSHIP_FEE* = 5.u256
+  MEMBERSHIP_FEE* = 1000000000000000.u256
   #  the current implementation of the rln lib only supports a circuit for Merkle tree with depth 32
```
Comment (s1fr0) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim:94]:
```suggestion
  #  the current implementation of the rln lib supports a circuit for Merkle tree with depth 20
```

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 3]
```diff
@@ -607,16 +611,19 @@ proc addAll*(rlnInstance: RLN[Bn256], list: seq[IDCommitment]): bool =
...
 
-proc subscribeToGroupEvents(ethClientUri: string, contractAddress: Address, blockNumber: string = "0x0", handler: RegistrationEventHandler) {.async, gcsafe.} = 
+proc subscribeToGroupEvents(ethClientUri: string, ethAccountAddress: Address, contractAddress: Address, blockNumber: string = "0x0", handler: RegistrationEventHandler) {.async, gcsafe.} = 
   ## connects to the eth client whose URI is supplied as `ethClientUri`
   ## subscribes to the `MemberRegistered` event emitted from the `MembershipContract` which is available on the supplied `contractAddress`
   ## it collects all the events starting from the given `blockNumber`
   ## for every received event, it calls the `handler`
   
   # connect to the eth client
-  let web3 = await newWeb3(ETH_CLIENT)
```
Comment (s1fr0) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:617]: If `ETH_CLIENT` is no more needed (replaced by input `ethClientUri`) then probably you can remove https://github.com/status-im/nwaku/blob/993781a137a6ce63d3be4c7cd2d9e8bd5b403c98/waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim#L96-L98

**Review by s1fr0 — APPROVED** [review]
Looks good!

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim (hunk 3 above)
Comment (staheri14) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:617]: It is used for the testing purposes.

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim (hunk 2 above)
Comment (staheri14) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim]: Thanks, fixed in d5c60f2aa16db9cba15df40dd60171f1318531ef

**Pull request merged**